benchmarks/helics/BenchmarkMain.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright (c) 2017-2020,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC.  See
+the top-level NOTICE for additional details. All rights reserved.
+SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "BenchmarkFederate.hpp"
+#include "EchoHubFederate.hpp"
+#include "EchoLeafFederate.hpp"
+#include "EchoMessageHubFederate.hpp"
+#include "EchoMessageLeafFederate.hpp"
+#include "MessageExchangeFederate.hpp"
+#include "PholdFederate.hpp"
+#include "RingTransmitFederate.hpp"
+#include "RingTransmitMessageFederate.hpp"
+#include "TimingHubFederate.hpp"
+#include "TimingLeafFederate.hpp"
+#include "helics/core/helicsCLI11.hpp"
+
+#include <chrono>
+#include <iostream>
+
+std::shared_ptr<BenchmarkFederate> fed;
+
+template<class T>
+void addBM(helics::helicsCLI11App& app, std::string name, std::string description)
+{
+    app.add_subcommand(name, description)
+        ->callback([]() { fed = std::make_shared<T>(); })
+        ->footer([] {
+            T().initialize("", "--help");
+            return std::string{};
+        });
+}
+
+int main(int argc, char* argv[])
+{
+    helics::helicsCLI11App app(
+        "HELICS benchmark federates for use in multinode benchmark setups", "helics_benchmarks");
+    app.ignore_case()->prefix_command();
```

Comment:
might want `->ignore_underscore()` here as well so you can do "echo_hub" as a subcommand and others as well.